image-loader/app/controllers/ImageLoaderController.scala

```diff
@@ -170,6 +179,8 @@ class ImageLoaderController(auth: Authentication,
         .map {
           r => {
             logger.info("importImage request end")
+            uploadStatusTable.setStatus(UploadStatus(uri.split("/").last, uploadedBy.getOrElse(FALLBACK),
+              filename.getOrElse(FALLBACK), UploadStatus.COMPLETED))
```

Comment:
This should be a partial update - we only need to update the status here, not overwrite the other fields.

This only deals with the happy path. If the import fails we do not update the `UploadStatus`.

You might want to do this in an `onComplete` so that both the success and failure cases can be dealt with in a single place. However the `.recover` section converts `Failure` back into `Success` so it would need to be before that.

Remember also that we need to let the user know _what_ went wrong. As such, the UploadStatus needs an optional error message.